lupa/_lupa.pyx

```diff
@@ -819,6 +829,41 @@ cdef class _LuaTable(_LuaObject):
             lua.lua_settop(L, old_top)
             unlock_runtime(self._runtime)
 
+    @cython.final
+    cdef tuple _unpack(self):
+        """Unpacks table into a tuple of positional arguments
+        and a dictionary of keyword arguments
+        """
+        assert self._runtime is not None
+        cdef tuple args
+        cdef dict kwargs = {}
+        cdef bytes encoding
+        cdef Py_ssize_t index, length
+        lock_runtime(self._runtime)
+        try:
+            encoding = self._runtime._source_encoding
```

Comment:
Wow, this is literally the only thing that we need the `LuaRuntime` for here. Isn't there another way to get the encoding than to put the `LuaRuntime` in the Lua registry?

Reply:
We need the `LuaRuntime` for building the `_LuaTable` and converting its keys and values. If we opt to write a `LuaRuntime`-less `_unpacks`, we'd have to duplicate a lot of code.